libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2969,6 +2972,17 @@ bool GCS_MAVLINK::try_send_message(const enum ap_message id)
         send_vibration();
         break;
 
+#ifdef HAVE_AP_BLHELI_SUPPORT
+    case MSG_ESC_TELEMETRY: {
+        CHECK_PAYLOAD_SIZE(ESC_TELEMETRY_1_TO_4);
```

Comment:
You should omit this, and have `send_esc_telemetry_mavlink` return false if it failed to get a message sent (otherwise you can end up without the backpressure causing us to slow streams enough to get the later messages we wanted to send emitted. Not a critical thing since we get this wrong in a number of locations.

Reply:
it is bad either way. If we return false then the next call will send the 1_TO_4 msg again